examples/cifar/train_cifar.py

```diff
@@ -92,7 +103,11 @@ def main():
          'main/accuracy', 'validation/main/accuracy', 'elapsed_time']))
 
     # Print a progress bar to stdout
-    trainer.extend(extensions.ProgressBar())
+    # NOTE: If you use the EarlyStoppingTrigger,
+    #       training_length is needed to set
+    #       because trainer.stop_trigger is not normal interval trigger.
+    trainer.extend(extensions.ProgressBar(
+        training_length=(args.epoch, 'epoch')))
```

Comment:
@beam2d  san.

I have a question.
If we use `EarlyStoppingTrgigger` as `stop_trigger`, we have to specify the `training_length` here.
Without `training_length`, it may raise TypeError because `EarlyStoppingTrgigger` 
 is not [IntervalTrgger](https://github.com/chainer/chainer/blob/v3.1.0/chainer/training/extensions/progress_bar.py#L47).
We can avoid this error to specify the training_length which value is `max_epoch`.

Are there some good way which we do not have to specify `training_length` manually?
And should I write documentations about this behavior in EarlyStoppingTrigger?

Reply:
The problem is that the way `Trainer` detects the training length is not Pythonic; it uses `isinstance`. It should detect the training length by duck typing so that `EarlyStoppingTrigger` can support that protocol without hacking the class hierarchy. I think it should be done in another PR so that this PR can concentrate purely on adding `EarlyStoppingTrigger`.

Reply:
If you want, it's ok to do it in this PR as well (though it will then take more time to merge).

Reply:
I understood.
I agree with you and will create separate PR to modify EarlyStoppingTrigger.